AttributeError: 'MypyFile' object has no attribute 'type'

Python 3.5.1, mypy e360dae.

Input file:

```python
__builtins__['asd'] = 'qwe'
```

Error:
```
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/runpy.py", line 170, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/me/projects/mypy/mypy/__main__.py", line 5, in <module>
    main(None)
  File "/Users/me/projects/mypy/mypy/main.py", line 54, in main
    type_check_only(sources, bin_dir, options)
  File "/Users/me/projects/mypy/mypy/main.py", line 98, in type_check_only
    python_path=options.python_path)
  File "/Users/me/projects/mypy/mypy/build.py", line 208, in build
    result = manager.process(initial_states)
  File "/Users/me/projects/mypy/mypy/build.py", line 408, in process
    next.process()
  File "/Users/me/projects/mypy/mypy/build.py", line 912, in process
    self.type_checker().visit_file(self.tree, self.tree.path)
  File "/Users/me/projects/mypy/mypy/checker.py", line 410, in visit_file
    self.accept(d)
  File "/Users/me/projects/mypy/mypy/checker.py", line 451, in accept
    typ = node.accept(self)
  File "/Users/me/projects/mypy/mypy/nodes.py", line 564, in accept
    return visitor.visit_assignment_stmt(self)
  File "/Users/me/projects/mypy/mypy/checker.py", line 1125, in visit_assignment_stmt
    self.check_assignment(s.lvalues[-1], s.rvalue, s.type is None)
  File "/Users/me/projects/mypy/mypy/checker.py", line 1170, in check_assignment
    self.check_indexed_assignment(index_lvalue, rvalue, rvalue)
  File "/Users/me/projects/mypy/mypy/checker.py", line 1505, in check_indexed_assignment
    self.try_infer_partial_type_from_indexed_assignment(lvalue, rvalue)
  File "/Users/me/projects/mypy/mypy/checker.py", line 1519, in try_infer_partial_type_from_indexed_assignment
    if var is not None and isinstance(var.type, PartialType):
AttributeError: 'MypyFile' object has no attribute 'type'

*** INTERNAL ERROR ***

a.py:1: error: Internal error -- please report a bug at https://github.com/JukkaL/mypy/issues
```

--- a/mypy/checker.py
+++ b/mypy/checker.py
@@ -1528,7 +1528,7 @@ class TypeChecker(NodeVisitor[Type]):
     def try_infer_partial_type_from_indexed_assignment(
             self, lvalue: IndexExpr, rvalue: Node) -> None:
         # TODO: Should we share some of this with try_infer_partial_type?
-        if isinstance(lvalue.base, RefExpr):
+        if isinstance(lvalue.base, RefExpr) and isinstance(lvalue.base.node, Var):
             var = cast(Var, lvalue.base.node)
             if var is not None and isinstance(var.type, PartialType):
                 type_type = var.type.type

--- a/mypy/test/data/check-inference.test
+++ b/mypy/test/data/check-inference.test
@@ -1696,3 +1696,12 @@ def f(): pass
 a = {0: [0]} if f() else {0: []}
 a() # E: Dict[int, List[int]] not callable
 [builtins fixtures/dict.py]
+
+[case testMisguidedSetItem]
+from typing import Generic, Sequence, TypeVar
+T = TypeVar('T')
+class C(Sequence[T], Generic[T]): pass
+C[0] = 0
+[out]
+main:4: error: Type expected within [...]
+main:4: error: Unsupported target for indexed assignment

